src/math_parser_diag.h

```diff
@@ -141,6 +142,7 @@ inline std::map<std::string_view, dialogue_func_eval> const dialogue_eval_f{
     { "field_strength", { "ung", 1, field_strength_eval } },
     { "game_option", { "g", 1, option_eval } },
     { "has_trait", { "un", 1, has_trait_eval } },
+    { "knows_proficiency", { "un", 1, knows_proficiency_eval } },
```

Comment:
```suggestion
    { "has_trait", { "un", 1, has_trait_eval } },
    { "has_proficiency", { "un", 1, knows_proficiency_eval } },
```
Please use the same keyword as for the non-math condition, just to avoid confusion. I've tried it with your EOC sample and it worked fine.
![Screenshot from 2023-11-29 07-41-31](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/a0c8abfe-4a0d-49e2-87d6-6a0c91418f5d)